x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/utils/NamedXContentObjectHelper.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ml.utils;
+
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.List;
+
+public final class NamedXContentObjectHelper {
```

Comment:
Just to make sure: to what extent do we want to decouple server code from client code?
It feels to me that a utility class like this one could be placed in some "common" package visible by both server and client.

Reply:
Potentially, but I am not convinced it belongs in the root server code. It is something only used by this plugin.

Reply:
Ok, got it.